src/main/java/com/netflix/msl/entityauth/X509AuthenticationData.java

```diff
@@ -116,7 +130,7 @@ public X509AuthenticationData(final X509Certificate x509cert) throws MslCryptoEx
      * @return the X.509 certificate.
      */
     public X509Certificate getX509Cert() {
-        return x509cert;
+    	return certs.iterator().next();
```

Comment:
This behavior makes the class non-thread-safe and ambiguous. The function name should be changed if you want one function to return both x509certificate and x509chain. It might be better to have two getters, and one of them will return null.